File: javabot-play/app/utils/ChangeDao.scala
package utils

import javabot.model.Change
import javabot.model.criteria.ChangeCriteria
import models.ChangeForm
import scala.collection.JavaConversions._

class ChangeDao extends javabot.dao.ChangeDao {
  def find(form: ChangeForm, start: Int, count: Int): (Long, Seq[Change]) = {
    val criteria = new ChangeCriteria(ds)
    form.message.map( message => criteria.message().contains(message))
    criteria.changeDate().order(false)

    val query = criteria.query()
    val total = ds.getCount(query)
    query.offset(start)
    query.limit(count)
    Pair(total, query.asList().toSeq)
  }
}

File: web/app/utils/ChangeDao.java
package utils;

import java.util.List;

import javabot.model.Change;
import javabot.model.criteria.ChangeCriteria;
import org.mongodb.morphia.query.Query;
import play.libs.F.Tuple;

public class ChangeDao extends javabot.dao.ChangeDao {
  public Tuple<Long, List<Change>> find(Change change, int start, int count) {
    ChangeCriteria criteria = new ChangeCriteria(ds);
    if(change.getMessage() != null) {
      criteria.message().contains(change.getMessage());
    }
    criteria.changeDate().order(false);

    Query<Change> query = criteria.query();
    long total = ds.getCount(query);
    query.offset(start);
    query.limit(count);
    return new Tuple<>(total, query.asList());
  }

}
